app/grandchallenge/evaluation/views/api.py

```diff
@@ -23,3 +46,118 @@ class EvaluationViewSet(ReadOnlyModelViewSet):
         *api_settings.DEFAULT_RENDERER_CLASSES,
         PaginatedCSVRenderer,
     )
+
+    @action(
+        detail=True, methods=["PATCH"], permission_classes=[CanClaimEvaluation]
+    )
+    def claim(self, request, *args, **kwargs):
+        evaluation = self.get_object()
+
+        if not evaluation.status == Evaluation.PENDING:
+            return Response(
+                {"status": "You can only claim pending evaluations."},
+                status=400,
+            )
+
+        if request.user.claimed_evaluations.exists():
+            return Response(
+                {"status": "You can only claim one evaluation at a time."},
+                status=400,
+            )
+
+        evaluation.status = Evaluation.CLAIMED
+        evaluation.claimed_by = request.user
+        evaluation.started_at = now()
+        evaluation.save()
+
+        serializer = ExternalEvaluationSerializer(
+            instance=evaluation, context={"request": request}
+        )
+        return Response(serializer.data)
+
+    @action(detail=False, methods=["GET"])
+    def claimable_evaluations(self, request, *args, **kwargs):
+        queryset = get_objects_for_user(
+            request.user, "evaluation.claim_evaluation"
+        ).filter(status=Evaluation.PENDING)
+        serializer = ExternalEvaluationSerializer(
+            queryset, many=True, context={"request": request}
+        )
+        return Response(serializer.data)
+
+    @action(
+        detail=True,
+        methods=["PATCH"],
+        permission_classes=[CanClaimEvaluation],
+        serializer_class=ExternalEvaluationUpdateSerializer,
+    )
+    def update_external_evaluation(self, request, *args, **kwargs):
+        evaluation = self.get_object()
+        if evaluation.status != Evaluation.CLAIMED:
+            return Response(
+                {
+                    "status": "You need to claim an evaluation before you can update it."
+                },
+                status=400,
+            )
+
+        if request.user != evaluation.claimed_by:
+            return Response(
+                {
+                    "status": "You do not have permission to update this evaluation."
+                },
+                status=403,
+            )
+
+        if (
+            evaluation.started_at - now()
+        ).seconds > settings.EXTERNAL_EVALUATION_TIMEOUT_IN_SECONDS:
+            evaluation.status = Evaluation.CANCELLED
+            evaluation.error_message = "External evaluation timed out."
+            evaluation.save()
```

Comment:
We shouldn't modify the object and return a status of 400. Again we have the problem of the notifications. I think it's best to call the `cancel_external_evaluations_past_timeout` task here instead and return 400.